src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -320,21 +324,33 @@ export class RoslynLanguageServer {
 
     public async openSolution(solutionFile: vscode.Uri): Promise<void> {
         this._solutionFile = solutionFile;
-        await this.sendOpenSolutionNotification();
+        this._projectFiles = new Array<vscode.Uri>();
+        await this.sendOpenSolutionAndProjectsNotifications();
     }
 
-    private async sendOpenSolutionNotification(): Promise<void> {
-        if (
-            this._solutionFile !== undefined &&
-            this._languageClient !== undefined &&
-            this._languageClient.isRunning()
-        ) {
-            const protocolUri = this._languageClient.clientOptions.uriConverters!.code2Protocol(this._solutionFile);
-            await this._languageClient.sendNotification('solution/open', new OpenSolutionParams(protocolUri));
+    public async openProjects(projectFiles: vscode.Uri[]): Promise<void> {
```

Comment:
do we have a picker for projects like we do slns?

Reply:
I realize after I write this, a project picker for projects may not make sense, especially in huge workspaces.  Though I am curious - what happens if we load a project that depends on another project via project reference (and the dependency is not loaded)?

Reply:
@dibarbet I'm not including a picker _in this PR_. Omnisharp did have one but in the case of a large workspace I'm not entirely sure how useful that experience would have been, and at least one customer so far has pointed out they were switching projects when they probably didn't mean to.

If the dependency isn't there, we'll just treat it as a metadata reference and load the DLL.

Reply:
Gotcha.  Lets make sure that experience is tracked as well.  I agree it could be complicated - for example it would be nice if you once you select the projects to load you can save it somewhere (e.g. similar to default solution).  Or maybe instead we just implement load projects on demand... Not sure either.

Reply:
I'd hope projects-on-demand meets many needs, or yes, some other way to specify the list of projects (probably via globbing.)